### PR TITLE
More linter exception hardening and other fixes

### PR DESCRIPTION
**Pull request by StephenWeatherford**

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
image

File: src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs [hunk 1]
```diff
@@ -25,17 +25,24 @@ internal class LinterAnalyzer : IBicepAnalyzer
 
         private ConfigHelper configHelper = new ConfigHelper();
         private ImmutableArray<IBicepAnalyzerRule> RuleSet;
+        private ImmutableArray<IBicepAnalyzerDiagnostic> RuleCreationErrors;
+
+        internal const string FailedRuleCode = "Linter Rule Error";
```
Comment (jfleisher) [inline src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs]: this won't be a config code, but should we still style the code like the other rule codes with lower case and hyphens?

**Review by jfleisher — APPROVED** [review]
One nit question about the rule code. Looks good.

File: src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs (hunk 1 above)
Comment (StephenWeatherford) [reply src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs]:
```suggestion
        internal const string FailedRuleCode = "linter-internal-error";
```

Comment (StephenWeatherford) [reply src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs]: For later, we might consider making it into a bicep-coded error.

**Pull request merged**